File: Interface Exploration Base/Assets/Plugins/DS501/scripts/core/CSV.js
#pragma strict
/*
//TODO: convert to C#?
class CSV
{

	static function write_data( filename, array )
	{
		var out_file : StreamWriter;
		out_file = new File.AppendText( filename );
		//var timestamp = getTimestamp();
		var line;
		var didOne = false;
		
		//Debug.Log( filename );
		//Debug.Log( array );
		
		var s : String;
		for( var a in array )
		{
			//Debug.Log( a + " : " + a.GetType() );
			//if( a.GetType() != String && a.GetType() != Number && a.GetType() != float )
			if( a.GetType() != String )
				s = a.ToString();
			else
				s = a;
			//Debug.Log( s );
		
			if( !didOne )		didOne = true;
			else				line += ", ";
			line += s;
		}
		out_file.WriteLine( line );
		out_file.Close();
	}
}
*/

File: Interface Exploration Base/Assets/Plugins/DS501/scripts/core/CSV.cs
using UnityEngine;
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;

public class CSV
{
	public static void log<T>(IList<T>array)
	{
		string filename = "log.txt";
		Int32 unixTimestamp = (Int32)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1))).TotalSeconds;

		if (!File.Exists (filename)) {
			File.Create (filename);
		}

		int i = 0;
		string[] data = new string[array.Count + 1];
		data [0] = unixTimestamp.ToString ();
		foreach (T item in array) {
			data[i + 1] = item.ToString ();
			i++;
		}

		using (StreamWriter sw = File.AppendText(filename)) 
		{
			string line = String.Join (",", data);
			sw.WriteLine(line);
		}
	}
}
